lib/elixir/src/elixir_expand.erl

```diff
@@ -357,10 +357,15 @@ expand({Name, Meta, Kind} = Var, #{vars := Vars} = E) when is_atom(Name), is_ato
         {var, true} ->
           form_error(Meta, ?key(E, file), ?MODULE, {undefined_var, Name, Kind});
         _ ->
-          Message =
-            io_lib:format("variable \"~ts\" does not exist and is being expanded to \"~ts()\","
-              " please use parentheses to remove the ambiguity or change the variable name", [Name, Name]),
-          elixir_errors:warn(?line(Meta), ?key(E, file), Message),
+          case ?key(E, match_vars) of
+            warn ->
+              Message =
+                io_lib:format("variable \"~ts\" does not exist and is being expanded to \"~ts()\","
+                  " please use parentheses to remove the ambiguity or change the variable name", [Name, Name]),
+              elixir_errors:warn(?line(Meta), ?key(E, file), Message);
+            apply ->
+              ok
+          end,
```

Comment:
The implementation is sound but the idea is debatable. I have tried the following approaches:

  1. Autocompletion of nullary calls - it is a bad user experience because it works only some times (for example, it wouldn't work for `open/0` and `whereami/0` because we also have other arities)
  2. Parens autocompletion of all calls - if we add only `(`, we get incomplete input most of the times. If we add both, then we need to go back one position (currently it is not possible to add both and then put the cursor on the middle)
  3. Accept `continue` in the shell - the UX is great, the inconsistency is, well, inconsistent

We can bring the warning back once we figure out autocompletion but since this feature is about the user experience, I have put it first here.